tests/worker.test.ts

```diff
@@ -0,0 +1,133 @@
+        events: count,
+        concurrency: piscina.threads.length,
+        totalMs: ms,
+        averageEventMs: ms / count,
+    }
+
+    return log
+}
+
+function setupPiscina(workers: number, code: string, tasksPerWorker: number) {
+    return makePiscina({
+        ...defaultConfig,
+        WORKER_CONCURRENCY: workers,
+        TASKS_PER_WORKER: tasksPerWorker,
+        LOG_LEVEL: 'log',
+        __jestMock: mockJestWithIndex(code),
+    })
+}
+
+test('piscina worker test', async () => {
+    const coreCount = os.cpus().length
+
+    const workers = [1, 2, 4, 8].filter((cores) => cores <= coreCount)
+    const rounds = 5
+
+    const tests: { testName: string; events: number; testCode: string }[] = [
+        {
+            testName: 'simple',
+            events: 10000,
+            testCode: `
+                function processEvent (event, meta) {
+                    event.properties = { "somewhere": "over the rainbow" };
+                    return event
+                }
+            `,
+        },
+        {
+            testName: 'for200k',
+            events: 10000,
+            testCode: `
+                function processEvent (event, meta) {
+                    let j = 0; for(let i = 0; i < 200000; i++) { j = i };
+                    event.properties = { "somewhere": "over the rainbow" };
+                    return event
+                }
+            `,
+        },
+        {
+            testName: 'timeout100ms',
+            events: 5,
+            testCode: `
+                async function processEvent (event, meta) {
+                    await new Promise(resolve => __jestSetTimeout(() => resolve(), 100))
+                    event.properties = { "somewhere": "over the rainbow" };
+                    return event             
+                }
+            `,
+        },
+    ]
+
+    const results: Array<Record<string, string | number>> = []
+    for (const { testName, events, testCode } of tests) {
+        const result: Record<string, any> = {
+            testName,
+            coreCount,
+        }
+        for (const cores of workers) {
+            const piscina = setupPiscina(cores, testCode, 1)
+
+            // warmup
+            await processCountEvents(cores * 4, piscina)
+
+            // start
+            let throughput = 0
+            for (let i = 0; i < rounds; i++) {
+                const { eventsPerSecond } = await processCountEvents(events, piscina)
+                throughput += eventsPerSecond
+            }
+            result[`${cores} cores`] = Math.round(throughput / rounds)
+            await piscina.destroy()
+        }
+        results.push(result)
+        console.log(JSON.stringify({ result }, null, 2))
+    }
+    console.table(results)
+    // expect that adding more cores (up to coreCount) increases throughput
+    // for (let i = 1; i < workers.length; i++) {
+    //     expect(result[`${workers[i - 1]} cores`]).toBeLessThan(result[`${workers[i]} cores`])
+    // }
```

Comment:
I may be wrong, but this looks like commented out dead cod.

Reply:
RIP